binary_search_tree/tree.py

```diff
@@ -8,41 +8,124 @@ def __init__(self, key, val = None):
         self.left = None
         self.right = None
         
-
-
 class Tree:
     def __init__(self):
         self.root = None
 
-    # Time Complexity: 
-    # Space Complexity: 
+    def add_helper(self, current_root, new_node):
+        if current_root is None:
+            return new_node
+        
+        if new_node.key < current_root.key:
+            current_root.left = self.add_helper(current_root.left, new_node)
+
+        else:
+            current_root.right = self.add_helper(current_root.right, new_node)
+
+        return current_root
+
+    def add(self, key, value):
+        if self.root is None:
+            self.root = TreeNode(key, value)
+            return
+
+        new_node = TreeNode(key, value)
+        self.add_helper(self.root, new_node)
```

Comment:
Looks like this was an older recursive solution. Good on you for solving it both ways! And correctly at that!

However, do be careful about leaving old unused versions of code in your source code, whether they are commented out or not. Especially if they have have the same name as they do in this case. Just mainly a style thing about aiming to keep your code clean of clutter.